packages/blockchain/celo/src/lib/services/celo.kms.ts

```diff
@@ -0,0 +1,51 @@
+import { EvmBasedWeb3 } from '@tatumio/shared-blockchain-evm-based'
+import { ChainTransactionKMS, EvmBasedBlockchain } from '@tatumio/shared-core'
+import { PendingTransaction, ApiServices } from '@tatumio/api-client'
+import { abstractBlockchainKms } from '@tatumio/shared-blockchain-abstract'
+
+export const celoKmsService = (args: { blockchain: EvmBasedBlockchain; web3: EvmBasedWeb3 }) => {
+  return {
+    ...abstractBlockchainKms(args),
+    // async sign(tx: ChainTransactionKMS, fromPrivateKey: string, provider?: string): Promise<string> {
+    //   // @TODO: probably bug in OpenAPI
+    //   ;(tx as PendingTransaction).chain = 'CELO' as any
+
+    //   const client = args.web3.getClient(provider, fromPrivateKey)
+    //   const transactionConfig = JSON.parse(tx.serializedTransaction)
+    //   transactionConfig.gas = await client.eth.estimateGas(transactionConfig)
+
+    //   if (!transactionConfig.nonce) {
+    //     transactionConfig.nonce = await ApiServices.blockchain.eth.ethGetTransactionCount(
+    //       client.eth.defaultAccount as string,
+    //     )
+    //   }
+    //   if (
+    //     !transactionConfig.gasPrice ||
+    //     transactionConfig.gasPrice === '0' ||
+    //     transactionConfig.gasPrice === 0 ||
+    //     transactionConfig.gasPrice === '0x0'
+    //   ) {
+    //     transactionConfig.gasPrice = await args.web3.getGasPriceInWei()
+    //   }
+    //   return (await client.eth.accounts.signTransaction(transactionConfig, fromPrivateKey as string))
+    //     .rawTransaction as string
+    // export const signKMSTransaction = async (tx: ChainTransactionKMS, fromPrivateKey: string, provider?: string) => {
```

Comment:
Is it possible to use same implementation as ETH? Or should I use implementation from `feature/submodule` which uses `CeloWallet` lib?

Reply:
I guess you need CeloWallet one